corehq/ex-submodules/pillowtop/utils.py

```diff
@@ -172,11 +187,19 @@ def get_pillow_json(pillow_config):
         seconds_since_last = 0
         prettified_time_since_last = ''
         hours_since_last = None
-    offsets = pillow.get_change_feed().get_latest_offsets_json()
+
+    if _is_kafka(checkpoint):
+        # breaking composition boundaries so that we can use one Kafka connection
+        offsets = {
+            "{},{}".format(tp.topic, tp.partition): offset
+            for tp, offset in consumer.end_offsets(checkpoint.wrapped_sequence.keys()).items()
```

Comment:
@emord modernize wants to wrap ```checkpoint.wrapped_sequence.keys()``` with ```list()```.  Is ```checkpoint.wrapped_sequence``` a dict?  If so I'll go with ```six.iterkeys()```.

Reply:
it appears that it is a dict. I think it should be fine to go with `list` to be safe though